cpp-ch/local-engine/Functions/SparkParseURL.cpp

```diff
@@ -380,6 +380,21 @@ struct SparkExtractURLHost
         if (userinfo_delim_pos && userinfo_delim_pos < end)
```

Comment:
instead of parsing url manually, why not use `Poco::URI`?  It should be much easier than current implementation.

Reply:
The implementation of `SparkParseURL.cpp` is seems try to use the url parsing function of clickhouse,  and I think it is may be better than `Poco::URI`.

Reply:
But the existing code looks a bit messy and will be difficult to maintain in the future. Would you mind investigating the feasibility of Poco::URI?

Reply:
Besides, make sure that all related spark uts are enabled.